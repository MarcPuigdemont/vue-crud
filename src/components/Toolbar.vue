<template>
  <div class="d-flex justify-content-between">
    <b-button pill class="add_button" data-test="new-airline-button" @click="$emit('create')"><i class="material-icons">add</i></b-button>
    <div class="d-flex flex-direction-row">
      <i
        class="material-icons filter-button"
        :disabled="!filterServices.bags"
        title="Filter by Bags service available"
        data-test="filter-bags"
        @click="toggleService('bags')"
      >
        {{ icons['BAGS'] }}
      </i>
      <i
        class="material-icons filter-button"
        :disabled="!filterServices.checkin"
        title="Filter by Check-in service available"
        data-test="filter-checkin"
        @click="toggleService('checkin')"
      >
        {{ icons['CHECKIN'] }}
      </i>
      <i
        class="material-icons filter-button"
        :disabled="!filterServices.seats"
        title="Filter by Seats service available"
        data-test="filter-seats"
        @click="toggleService('seats')"
      >
        {{ icons['SEATS'] }}
      </i>
      <input
        v-model="searchFilter"
        class="form-control filter-form-search"
        type="search"
        placeholder="Search/Filter"
        aria-label="Search/Filter"
        data-test="filter-input"
      />
    </div>
  </div>
</template>
<script>
import constants from '@/constants';

export default {
  data() {
    return {
      searchFilter: '',
      icons: constants.ICONS,
      filterServices: {
        bags: false,
        checkin: false,
        seats: false,
      },
    };
  },
  methods: {
    toggleService(service) {
      this.filterServices[service] = !this.filterServices[service];
      this.$emit('service-filter', this.filterServices);
    },
  },
  watch: {
    searchFilter: function(val) {
      this.$emit('filter', val);
    },
  },
};
</script>
<style scoped>
.add_button {
  height: 38px;
  width: 38px;
  position: relative;
  outline: none !important;
  box-shadow: none !important;
  background-color: #293041;
  border-color: #293041;
}
.add_button i {
  position: absolute;
  top: 6px;
  left: 6px;
}
.filter-form-search {
  max-width: 200px;
}
.filter-button {
  margin-top: 0.5rem;
  margin-right: 2px;
  margin-left: 2px;
  cursor: pointer;
}
</style>
