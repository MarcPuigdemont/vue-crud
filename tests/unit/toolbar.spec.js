import { mount, shallowMount } from '@vue/test-utils';
import Toolbar from '@/components/Toolbar.vue';

describe('Toolbar', () => {
  test('should match the snapshot', () => {
    const wrapper = mount(Toolbar);
    expect(wrapper.element).toMatchSnapshot();
  });
  test('should emit proper services', () => {
    const wrapper = shallowMount(Toolbar, {});

    wrapper.find('[data-test=filter-bags]').element.click();
    let services = { bags: true, checkin: false, seats: false };
    expect(wrapper.emitted('service-filter')[0][0]).toEqual(services);

    wrapper.find('[data-test=filter-checkin]').element.click();
    services = { bags: true, checkin: true, seats: false };
    expect(wrapper.emitted('service-filter')[1][0]).toEqual(services);

    wrapper.find('[data-test=filter-seats]').element.click();
    services = { bags: true, checkin: true, seats: true };
    expect(wrapper.emitted('service-filter')[2][0]).toEqual(services);
  });
  test('should emit proper search filter', () => {
    const wrapper = shallowMount(Toolbar, {});
    wrapper.find('[data-test=filter-input]').setValue('1234');
    expect(wrapper.emitted('filter')[0][0]).toEqual('1234');
  });
});
