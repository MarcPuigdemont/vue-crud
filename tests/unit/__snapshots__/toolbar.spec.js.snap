// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should match the snapshot 1`] = `
<div
  class="d-flex justify-content-between"
>
  <button
    class="btn add_button btn-secondary rounded-pill"
    data-test="new-airline-button"
    type="button"
  >
    <i
      class="material-icons"
    >
      add
    </i>
  </button>
   
  <div
    class="d-flex flex-direction-row"
  >
    <i
      class="material-icons filter-button"
      data-test="filter-bags"
      disabled="disabled"
      title="Filter by Bags service available"
    >
      
      work
    
    </i>
     
    <i
      class="material-icons filter-button"
      data-test="filter-checkin"
      disabled="disabled"
      title="Filter by Check-in service available"
    >
      
      where_to_vote
    
    </i>
     
    <i
      class="material-icons filter-button"
      data-test="filter-seats"
      disabled="disabled"
      title="Filter by Seats service available"
    >
      
      airline_seat_recline_normal
    
    </i>
     
    <input
      aria-label="Search/Filter"
      class="form-control filter-form-search"
      data-test="filter-input"
      placeholder="Search/Filter"
      type="search"
    />
  </div>
</div>
`;
